packages/sql-drizzle/src/internal/drizzle-patch.types.ts

```diff
@@ -0,0 +1,10 @@
+import type { SqlError } from "@effect/sql/Error"
+import type * as Effect from "effect/Effect"
+/**
+ * QueryPromise class is used for every Drizzle QueryBuilders, it's what allows Builders to be awaitable
+ * So since we need all QueryBuilders to also be effectable, we patch it's interface to extend Effect
+ * We don't however monkey patch the QueryPromise class itself, but the QueryBuilders that extend it to be able to attach the right client to them
```

Comment:
Patching QueryPromise might be cleaner. Is it exposed?

Reply:
yes, it's exposed

Reply:
But patching QueryPromise with dependency injecting the client will patch all instances (not just the one we declared), it's indeed doable, but @mikearnaldi has concerns about patching globally the types, so patching the propotype globally would be even more intrusive.

Reply:
> But patching QueryPromise with dependency injecting the client will patch all instances (not just for the client we declared), it's indeed doable, but @mikearnaldi has concerns about patching globally the types, so patching the propotype globally would be even more intrusive.

I don't really have concerns if the patching is valid, the issue would be patching instances globally that can't be used in an effect environment, if the patch adds `Effect<X, SqlError, Client>` it is valid because the client will need to be provided so as long as augmentation and patching are done at the same time it should work

Reply:
So i tested it, and patching QueryPromise does not work. No idea why, i get this error :
```md
Unknown Error: TypeError: yield* (intermediate value)(intermediate value)(intermediate value)(intermediate value)(intermediate value)(intermediate value) is not iterable
 ❯ test/sqlite.test.ts:31:43
     29|       const inserted = yield* db.insert(users).values({ name: "Alice" }).returning()
     30|       assert.deepStrictEqual(inserted, [{ id: 1, name: "Alice" }])
     31|       const selected = yield* db.select().from(users)
       |                                           ^
```
have you ever seen it ? If no solution, i'll keep the patching at builder level.

Reply:
I have another idea, will try it out and report back if it works.

Reply:
Updated #2860